app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/ReviewRegistrationController.scala

```diff
@@ -60,22 +62,28 @@ class ReviewRegistrationController @Inject() (
 
   def displayPage(): Action[AnyContent] =
     (authenticate andThen journeyAction).async { implicit request =>
-      if (request.registration.isCheckAndSubmitReady)
-        markRegistrationAsReviewed().map { _ =>
-          Ok(
-            reviewRegistrationPage(registration = request.registration,
-                                   liabilityStartLink = startRegistrationController.startLink
-            )
-          )
-        }
-      else
-        Future(Redirect(routes.TaskListController.displayPage()))
+      if (request.registration.isCheckAndSubmitReady) {
+        val reg: Registration = removePartialGroupMembers(request.registration)
```

Comment:
I did not consider that as I wanted to follow the single responsibility principle for that function.

Reply:
i guess what I'm really trying to avoid is having two registrations in play. Both are passed into `markRegistrationAsReviewed`. Maybe rename and do both updates on the one registration?

Reply:
Refactor this just before you add your comment.